### PR TITLE
[4.0] Category table parameterized queries

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds parameterized queries to `Joomla\CMS\Table\Category`.

### Testing Instructions

Simple queries, code review is fine.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 257e7e20f572e0a56f4027da610908c2ea27666d<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28279">issues.joomla.org/tracker/joomla-cms/28279</a>.</sub>

File: libraries/src/Table/Category.php [hunk 1]
```diff
@@ -95,7 +96,8 @@ protected function _getAssetParentId(Table $table = null, $id = null)
 			$query = $this->_db->getQuery(true)
 				->select($this->_db->quoteName('asset_id'))
 				->from($this->_db->quoteName('#__categories'))
-				->where($this->_db->quoteName('id') . ' = ' . $this->parent_id);
+				->where($this->_db->quoteName('id') . ' = :parentId')
+				->bind(':parentId', $parentId, ParameterType::INTEGER);
```
Comment (HLeithner) [inline libraries/src/Table/Category.php]: > I have tested this item  successfully on [257e7e2](https://github.com/joomla/joomla-cms/commit/257e7e20f572e0a56f4027da610908c2ea27666d)
> 
> code review
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/28279](https://issues.joomla.org/tracker/joomla-cms/28279).

Sorry but you failed ;-) 

$parentId is not defined

Comment (alikon) [reply libraries/src/Table/Category.php]: oops

Comment (alikon) [inline libraries/src/Table/Category.php]:
```suggestion
				->bind(':parentId', $this->parent_id, ParameterType::INTEGER);
```

Comment (HLeithner) [reply libraries/src/Table/Category.php]: thx

Comment (SharkyKZ) [reply libraries/src/Table/Category.php]: Haha, I did it on purpose to see if anyone notices :sweat_smile:

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**